File: PFE.domain/PFE.domai/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PFE.domain.Entities
{
    public class Project
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        public string? ImageUrl { get; set; }

      //  [Required]
        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Relations
        public virtual ICollection<Sprint> Sprints { get; set; } = new List<Sprint>();

        public virtual ICollection<ProjectUser> ProjectUsers { get; set; } = new List<ProjectUser>();
    }
}

File: gestion-stagiaire/src/app/components/models/Project.ts
import { Sprint } from "./Sprint";
export interface Project {
    id?: number;
    title: string;
    description: string;
    imageUrl?: string;
    startDate?: Date;
    endDate?: Date;
    createdAt?: Date;
    sprints?: Sprint[];
    users?: any[];
    projectUsers?: any[];
    highlighted?: boolean;
  }
  